modules/shared/src/main/scala/org/tessellation/ext/collection/SetRefUtils.scala

```diff
@@ -0,0 +1,18 @@
+package org.tessellation.ext.collection
+
+import cats.effect.{Ref, Sync}
+import cats.syntax.functor._
+
+object SetRefUtils {
+  implicit class RefOps[F[_]: Sync, K, V](val ref: Ref[F, Set[K]]) {
+
+    def add(k: K): F[Unit] =
+      ref.modify(a => (a + k, ()))
```

Comment:
If you return `Unit` you can just use `ref.upddate(_ + k)`

Reply:
I can change it. It's a copy paste from what we had in constellation.